hdf5.spec

```diff
@@ -13,18 +14,16 @@
 # You need to recompile all users of HDF5 for each version change
 Name: hdf5
 Version: %{major}.%{minor}
-Release: 4%{?dist}
+Release: 5%{hdf5_sha}%{?dist}
```

Comment:
Please see other specfiles (daos.spec for example) about how a git hash is incorporated into a Release:.  We should standardise on `Makefile` code and specfile code for doing this conditionally.